dev/bots/test.dart

```diff
@@ -634,6 +635,34 @@ Future<void> _runFrameworkTests() async {
   });
 }
 
+Future<void> _runFirebaseTestLabTests() async {
+  // Firebase Lab tests take ~20 minutes per integration test,
+  // so only one test is run per shard. Therefore, there are as
+  // many shards available as there are integration tests in this list.
+  // If you add a new test, add a corresponding firebase_test_lab-#-linux
+  // to .cirrus.yml
+  final List<String> integrationTests = <String>[
+    'release_smoke_test',
+    'abstract_method_smoke_test',
+    'android_embedding_v2_smoke_test',
+  ];
+  final Map<String, ShardRunner> subshards = <String, ShardRunner>{};
+
+  final String firebaseScript = path.join(flutterRoot, 'dev', 'bots', 'firebase_testlab.sh');
+  final String integrationTestDirectory = path.join(flutterRoot, 'dev', 'integration_tests');
+
+  for (int index = 0; index < integrationTests.length; index += 1) {
+    final String integrationTestPath = path.join(integrationTestDirectory, integrationTests[index]);
+    subshards['$index'] = () => runCommand(
+      firebaseScript,
+      <String>[ integrationTestPath ],
+      workingDirectory: flutterRoot,
+    );
+  }
+
+  await selectSubshard(subshards);
+}
```

Comment:
It might make sense to do the trick we use elsewhere that verifies that the right number of shards are actually given (by having the last shard be named something like `5_last` or something, I forget exactly how we do it).

Up to you.